services/web/server/src/simcore_service_webserver/computation_subscribe.py

```diff
@@ -77,29 +77,28 @@ async def _handle_computation_running_progress(
 
 async def progress_message_parser(app: web.Application, data: bytes) -> bool:
     # update corresponding project, node, progress value
-    rabbit_message = ProgressRabbitMessage.parse_raw(data)
+    try:
+        rabbit_message = ProgressRabbitMessageNode.parse_raw(data)
+    except ValidationError:
+        rabbit_message = ProgressRabbitMessageProject.parse_raw(data)
 
```

Comment:
Q: Is there some other way to distinguish between these two use cases?

Reply:
Actually that's an issue I've faced in the passed and I did not have a good solution.

I think that now I found a good one.

```python
    rabbit_message = parse_raw_as(
        Union[ProgressRabbitMessageNode, ProgressRabbitMessageProject], data
    )
```

Reply:
I've also added a test for future reference